src/Disks/DiskEncrypted.cpp

```diff
@@ -289,6 +289,12 @@ std::unique_ptr<ReadBufferFromFileBase> DiskEncrypted::readFile(
     std::optional<size_t> read_hint,
     std::optional<size_t> file_size) const
 {
+    if (read_hint && *read_hint > 0)
+        read_hint = *read_hint + FileEncryption::Header::kSize;
+
+    if (file_size && *file_size > 0)
+        file_size = *file_size + FileEncryption::Header::kSize;
```

Comment:
Why this issue was not found in any tests?

Reply:
Looks like we almost never pass `file_size` explicitly and in delegate (e.g. in `DiskLocal`, [link](https://github.com/ClickHouse/ClickHouse/blob/fdea042991f2a020f6ed18d860a67dea8438403d/src/Disks/DiskLocal.cpp#L366-L368)) it's obtained correctly.